model/testresult/testresult.go

```diff
@@ -0,0 +1,81 @@
+package testresult
+
+import (
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/db/bsonutil"
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	collection = "testresults"
+)
+
+// TestResult contains test data for a task.
+type TestResult struct {
+	ID        bson.ObjectId `bson:"_id" json:"_id"`
+	Status    string        `json:"status" bson:"status"`
+	TestFile  string        `json:"test_file" bson:"test_file"`
+	URL       string        `json:"url" bson:"url,omitempty"`
+	URLRaw    string        `json:"url_raw" bson:"url_raw,omitempty"`
+	LogID     string        `json:"log_id,omitempty" bson:"log_id,omitempty"`
+	LineNum   int           `json:"line_num,omitempty" bson:"line_num,omitempty"`
```

Comment:
I think this means that each document will be 1 line of 1 test result log right? Wouldn't it be simpler/faster to have all lines be in an array? I might be misunderstanding

Reply:
This is a link to a part of a log. When you click on Evergreen self-tests results (e.g., https://evergreen.mongodb.com/test_log/59c53393e3c33151bf000356#L25) you get a link to the line of the log for that particular test. The document for that test looks like
```
                {
                        "status" : "pass",
                        "test_file" : "TestBuildUpdateStatus",
                        "log_id" : "59c53393e3c33151bf000356",
                        "line_num" : 25,
                        "exit_code" : 0,
                        "start" : 1506096019,
                        "end" : 1506096019
                }
```